### PR TITLE
[#170406954] nomad deploy can deploy nomad jobs

### PR DESCRIPTION
**Pull request by austinworks**
use for waiting on any batch job to complete
allow command line usage
add deploy script for services
update parameterized job implementation to mesh with batch implementation
version bump

### CODE REVIEW

**Comment by omada-jacob-smith** [comment]
Other than a few missed new lines, this seems like it does most of what it should. Certainly a lot of code here, wouldn't mind a code review, though I think we did cover some of it during one before?

File: tests/__init__.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import nomad_deploy
 from io import StringIO
 import os
+from .deployment import TestDeployBatch
```
[the remark this answers is not in the document]
Comment (austinworks) [reply tests/__init__.py]: yeah i was confused about how to format my module names for targetted test runs

[the remark this answers is not in the document]
Comment (jimmyatwork) [reply tests/__init__.py]: Ah ok. I think this should work if you'd like to run a specific set of tests:
```shell
docker-compose run nomad_deploy python -m unittest tests.deployment.TestDeployBatch
```

File: tests/deployment/__init__.py [hunk 2]
```diff
@@ -1,21 +1,110 @@
...
+
+    def open_template(self, rel_path):
+        script_dir = os.path.dirname(__file__) 
+        abs_file_path = os.path.join(script_dir, rel_path)
+        self.template_source = open(abs_file_path, "r")
+        return self.template_source
+
+    def test_successful_batch_evaluation(self):
+        template_source = self.open_template("test_data/successful_batch.j2")
+        template = templating.JobTemplate(dict(), template_source, self.job_vars())
```
[the remark this answers is not in the document]
Comment (austinworks) [reply tests/deployment/__init__.py]: sounds good, thanks!

File: tests/deployment/__init__.py [hunk 3]
```diff
@@ -1,21 +1,110 @@
...
+        self.assertEqual(Api().job_status('successful_service'), 'running')
+        self.assertEqual(Api().deployment_status(deployment_id), 'successful')
 
+    def test_failed_service(self):
+        template_source = self.open_template("test_data/failed_service.j2")
+        template = templating.JobTemplate(dict(), template_source, self.job_vars())
+        with self.assertRaises(Exception) as context:
+            deploy_service_job(template, show_job=True, wait_seconds=10)
+        time.sleep(11)
+        self.assertEqual(Api().job_status('failed_service'), 'pending')
```
[the remark this answers is not in the document]
Comment (austinworks) [reply tests/deployment/__init__.py]: I think the goal here is to demonstrate that both 
- the job raises an error, and
- the job does not then go on to be successful
the sleep is a bit of a weak strategy for that, but I think that's what I was going for

File: tests/deployment/__init__.py [hunk 4]
```diff
@@ -1,21 +1,110 @@
...
+    deploy_job("test_data/successful_parameterized_batch.nomad")
+    dispatched_job = dispatch_parameterized_job("successful_parameterized_batch", {})
+    self.assertEqual(dispatched_job.status, 'dead')
 
   def test_failure(self):
-    deploy_job("failed_parameterized_batch.nomad")
-    success = deployment.dispatch_parameterized_job("failed_parameterized_batch", {})
-    self.assertFalse(success)
+    deploy_job("test_data/failed_parameterized_batch.nomad")
+    with self.assertRaises(Exception) as context:
```
[the remark this answers is not in the document]
Comment (austinworks) [reply tests/deployment/__init__.py]: sounds good

File: tests/deployment/test_data/example_job.j2 [hunk 5]
```diff
@@ -0,0 +1,63 @@
+job "example_job" {
```
[the remark this answers is not in the document]
Comment (austinworks) [reply tests/deployment/test_data/example_job.j2]: deleted

File: nomad_deploy/__init__.py [hunk 6]
```diff
@@ -36,50 +35,31 @@ def render():
...
-def main():
-  parser = argparse.ArgumentParser(description="Render a nomad job file.")
+def service():
+  args = __parse_args()
+  data = load_data(args.data)
+  template = templating.JobTemplate(templating.secret, args.template[0], data)
+  deployment.deploy_service_job(template, target_groups=args.target_group, show_job=args.show_job)
+  logging.info("\nSuccess!")
+
+def __parse_args():
```
[the remark this answers is not in the document]
Comment (austinworks) [reply nomad_deploy/__init__.py]: ah i've been hearing about "dunder" methods from other folks and was wondering why you didn't use them.  Thanks for explaining it.

**Pull request merged**